### PR TITLE
[DOC] Enhanced RDoc for String

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Treats:
- #scan
- #hex
- #oct
- #crypt
- #ord
- #sum

### CODE REVIEW

**Comment by peterzhu2118** [comment]
It looks like there are merge conflicts with #5723. Could you rebase?

**Comment by BurdetteLamar** [comment]
> It looks like there are merge conflicts with #5723. Could you rebase?

I don't see any difference in the two passages, do you?  Also, I don't know rebase.

**Comment by peterzhu2118** [comment]
It looks like commit 148605a46674dd30f8c9f9d90bd0594a8ed2f43b (the first commit) is causing conflicts.

You can try resolving conflicts using GitHub's web editor (see screenshot below).

image

You could also use a git client such as [GitHub Desktop](https://desktop.github.com/), where you can do a rebase by going to "Branch" -> "Rebase Current Branch" -> Select "Master" -> Click "Rebase".

You could also rebase through the command line, by doing a `git pull --rebase https://github.com/ruby/ruby master` and following instructions in the command line to resolve conflicts (if there are any conflicts). Then you can do a force push by doing `git push --force`.

**Comment by BurdetteLamar** [comment]
Conflict resolved (I think).

File: string.c [hunk 1]
```diff
@@ -10023,16 +10031,20 @@ rb_str_scan(VALUE str, VALUE pat)
 
 /*
  *  call-seq:
- *     str.hex   -> integer
+ *    hex -> integer
  *
- *  Treats leading characters from <i>str</i> as a string of hexadecimal digits
+ *  Interprets the leading substring os +self+ as a string of hexadecimal digits
```
Comment (peterzhu2118) [inline string.c:10036]: I think there's a typo:

```suggestion
 *  Interprets the leading substring of +self+ as a string of hexadecimal digits
```

File: string.c [hunk 2]
```diff
@@ -10044,19 +10056,22 @@ rb_str_hex(VALUE str)
 
 /*
  *  call-seq:
- *     str.oct   -> integer
+ *    oct -> integer
+ *
+ *  Interprets the leading substring os +self+ as a string of octal digits
```
Comment (peterzhu2118) [inline string.c:10061]: Here too:

```suggestion
 *  Interprets the leading substring of +self+ as a string of octal digits
```

File: string.c [hunk 3]
```diff
@@ -9958,33 +9958,41 @@ scan_once(VALUE str, VALUE pat, long *start, int set_backref_str)
...
- *     print "\n"
+ *  - If the pattern contains no groups, each result is the
+ *    matched string, <code>$&</code>.
+ *  - If the pattern contains groups, each result is an array
+ *    containing one entry per group.
  *
- *  <em>produces:</em>
+ *  With no block given, returns an array of the results:
+ *
+ *    s = 'cruel world;
```
Comment (peterzhu2118) [inline string.c:9978]: Typo:

```suggestion
 *    s = 'cruel world'
```

File: doc/string/sum.rdoc [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+modulo <tt>2**n - 1</tt>:
+
+  'hello'.sum     # => 532
+  'hello'.sum(4)  # => 4
+  'hello'.sum(64) # => 532
+  'тест'.sum      # => 1405
+  'こんにちは'.sum  # => 2582
+
+This is not a particularly strong checksum.
+
```
Comment (peterzhu2118) [inline doc/string/sum.rdoc:12]: Nit: extra newline.

Comment (BurdetteLamar) [reply doc/string/sum.rdoc:12]: Fixed.

**Comment by BurdetteLamar** [comment]
I'm not happy to have so many obvious errors in this PR.  Will proof more carefully.

File: doc/string/sum.rdoc (hunk 4 above)
Comment (peterzhu2118) [reply doc/string/sum.rdoc:12]: I don't see this as fixed

**Comment by peterzhu2118** [comment]
Don't worry about mistakes and typos in PRs, it happens to all of us!

File: doc/string/sum.rdoc (hunk 4 above)
Comment (BurdetteLamar) [reply doc/string/sum.rdoc]: Now fixed, I think.  Earlier, I pushed and walked away, not seeing a rejection.

**Review by peterzhu2118 — APPROVED** [review]
Looks good!

**Pull request merged**